src/routes/error-router.ts

```diff
@@ -63,13 +69,27 @@ ErrorRouter.use((err: Error, req: Request, res: Response, next: NextFunction) =>
 		"Not Found": 404
 	};
 
-	const messages = {
-		[Errors.MISSING_JIRA_HOST]: "Session information missing - please enable all cookies in your browser settings.",
-		[Errors.IP_ALLOWLIST_MISCONFIGURED]: `The GitHub org you are trying to connect is currently blocking our requests. To configure the GitHub IP Allow List correctly, <a href="${envVars.GITHUB_REPO_URL}/blob/main/docs/ip-allowlist.md">please follow these instructions</a>.`
-	};
+	let message: string | null = null;
```

Comment:
What was the drive behind completely refactoring this? Can't we follow the same pattern and add a new error to the messages object and update the Errors enum? If we stick to this pattern then we don't need to declare ctaUrl and message as undefined variables.

Reply:
The reason for changing the original pattern was to add `ctaUrl`. The call to action URL is only for the MISSING_GITHUB_APP_NAME error. We can easily add a call to action link to any of the existing errors or new errors, in this way. 
I didnt want to add another `if condition` to add the `ctaUrl` for this error.

Reply:
You can easily add a call to action url and pass an additional value to the frontend **without** refactoring this entire file. Just add `MISSING_GITHUB_APP_NAME` to the object:

```
	const messages = {
		[Errors.MISSING_JIRA_HOST]: "Session information missing - please enable all cookies in your browser settings.",
		[Errors.IP_ALLOWLIST_MISCONFIGURED]: `The GitHub org you are trying to connect is currently blocking our requests. To configure the GitHub IP Allow List correctly, <a href="${envVars.GITHUB_REPO_URL}/blob/main/docs/ip-allowlist.md">please follow these instructions</a>.`
	};
  [Errors.MISSING_GITHUB_APP_NAME]: "message..."
```

Then ditch the ctaUrl object, define it in the render function, and hardcode the "Retry" text on the frontend you don't need to refactor everything.
- **Kill the ctaUrl object:** ctaUrl = { text: "Retry", action: createUrlWithQueryString(req, "/session")};
- **Set the url in the render function:** `ctaUrl: createUrlWithQueryString(req, "/session")`
- and change the anchor tag to: `<a class="aui-button aui-button-primary m-4" href="{{ctaUrl}}">Retry</a>`

Reply:
* Reverted back to the previous logic. 
* The new error message was added as a part of the same object.
* Used query parameters for handling the action of the retry button